python/insights_ansible_playbook_lib/_keygen.py

```diff
@@ -0,0 +1,214 @@
+
+    return result
+
+
+def _generate_keys() -> str:
+    """
+    Generate GPG keys into a temporary directory.
+    """
+    # Create a temporary directory to store the keys
+    gpg_tmp_dir = tempfile.mkdtemp(
+        dir=TEMPORARY_GPG_HOME_PARENT_DIRECTORY,
+        prefix=TEMPORARY_GPG_HOME_PARENT_DIRECTORY_PREFIX,
+    )
+    logger.debug(f"Generating GPG keys into {gpg_tmp_dir}.")
+
+    instructions_file = pathlib.Path(gpg_tmp_dir) / "keygen"
+    instructions_file.write_text(
+        """Key-Type: EDDSA
+Key-Curve: ed25519
+Subkey-Type: ECDH
+Subkey-Curve: cv25519
+Name-Real: insights-ansible-playbook-verifier test
+Expire-Date: 0
+%no-protection
+%commit
+"""
+    )
+    instructions_text = instructions_file.read_text().strip().replace("\n", "\\n")
+    logger.debug(
+        f"Keys generation instructions written to file {instructions_file}: {instructions_text}"
+    )
+
+    # Generate the keys in a temporary directory
+    _run_gpg_command(
+        [
+            "/usr/bin/gpg",
+            "--batch",
+            "--homedir",
+            gpg_tmp_dir,
+            "--generate-key",
+            f"{instructions_file}",
+        ]
+    )
+
+    return gpg_tmp_dir
+
+
+def _export_key_pair(gpg_tmp_dir: str, keys_path: str) -> None:
+    """
+    Export the generated key pair to the `keys_path` directory.
+
+    :param gpg_tmp_dir: The GPG home directory where the key pair was generated.
+    :param keys_path: The directory where the key pair should be exported.
+    """
+    _run_gpg_command(
+        [
+            "/usr/bin/gpg",
+            "--homedir",
+            gpg_tmp_dir,
+            "--export",
+            "--armor",
+            "--yes",
+            "--output",
+            f"{keys_path}/key.public.gpg",
+        ]
+    )
+
+    _run_gpg_command(
+        [
+            "/usr/bin/gpg",
+            "--homedir",
+            gpg_tmp_dir,
+            "--export-secret-keys",
+            "--armor",
+            "--yes",
+            "--output",
+            f"{keys_path}/key.private.gpg",
+        ]
+    )
+
+
+def _get_fingerprint(gpg_tmp_dir: str, keys_path: str) -> str:
+    """
+    Get the fingerprint of the generated key pair and write it to a file in the `keys_path` directory.
+
+    :param gpg_tmp_dir: The GPG home directory where the key pair was generated.
+    :param keys_path: The directory where the key pair is exported.
+    """
+    result = _run_gpg_command(
+        [
+            "/usr/bin/gpg",
+            "--homedir",
+            gpg_tmp_dir,
+            "--fingerprint",
+            "insights-ansible-playbook-verifier test",
+        ],
+        fingerprint=True,
+    )
+
+    # Extract the fingerprint from the output
+    match = re.search(r"^\s+([A-F0-9\s]+)", result.stdout, re.MULTILINE)
+    gpg_fingerprint = match.group(1).strip() if match else ""
+
+    # Write the fingerprint to a file
+    with open(f"{keys_path}/fingerprint.txt", "w") as fingerprint_file:
```

Comment:
To be consistent with the other two filenames, should we name this `key.fingerprint.txt`?

Non-blocking.

Reply:
Changed the name to `key.fingerprint.txt`